pkg/runtime/bulksub_resiliency.go

```diff
@@ -0,0 +1,63 @@
+package runtime
+
+import (
+	"context"
+	"sync/atomic"
+
+	"github.com/cenkalti/backoff/v4"
+	"github.com/pkg/errors"
+
+	"github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/dapr/pkg/resiliency"
+	"github.com/dapr/dapr/utils"
+)
+
+// ApplyBulkSubscribeResiliency applies resiliency support to bulk subscribe. It tries to filter
+// out the messages that have been successfully processed and only retries the ones that have failed
+func (a *DaprRuntime) ApplyBulkSubscribeResiliency(bulkSubCallData *bulkSubscribeCallData,
+	psm pubsubBulkSubscribedMessage, deadLetterTopic string, path string, policyDef *resiliency.PolicyDefinition,
+	rawPayload bool, envelope map[string]interface{},
+) (*[]pubsub.BulkSubscribeResponseEntry, error) {
+	bscData := *bulkSubCallData
+	ctx := bscData.ctx
+	var subscribedEntries atomic.Pointer[[]pubSubMessage]
+	subscribedEntries.Store(&psm.pubSubMessages)
+	policyRunner := resiliency.NewRunnerWithOptions(
+		ctx, policyDef, resiliency.RunnerOpts[*[]pubsub.BulkSubscribeResponseEntry]{
+			Accumulator: func(bsre *[]pubsub.BulkSubscribeResponseEntry) {
+				for _, v := range *bsre {
+					// add to main bulkResponses
+					index := (*bscData.entryIdIndexMap)[v.EntryId]
+					(*bscData.bulkResponses)[index].EntryId = v.EntryId
+					(*bscData.bulkResponses)[index].Error = v.Error
+				}
+				filteredPubSubMsgs := utils.Filter(*subscribedEntries.Load(), func(ps pubSubMessage) bool {
+					index := (*bscData.entryIdIndexMap)[ps.entry.EntryId]
+					return (*bscData.bulkResponses)[index].Error != nil
+				})
+				psm.pubSubMessages = filteredPubSubMsgs
+				psm.length = len(filteredPubSubMsgs)
```

Comment:
1. Since subscribedEntries load psm.pubSubMessages (`subscribedEntries.Store(&psm.pubSubMessages)`), which have already changed as in line above, so modified psm is loaded, instead of original psm. And, it is inside Accumulator, so it seems safe.
2. I am just thinking, if atomic.Pointer is still needed, since these modifications are getting done only inside Accumulator now.